File: DataScience/Python/missing_data.py
#Plantilla de preprocesado - datos faltantes

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Importar el dataset
dataset = pd.read_csv('Data.csv')
X=dataset.iloc[:,:-1].values
y=dataset.iloc[:,3].values

#Tratamiento de datos faltantes NaN por medio de sustituirlos por la media
from sklearn.preprocessing import Imputer
imputer = Imputer(missing_values = "NaN", strategy = "mean", axis = 0) #axis = 0 es por columna, axis=1 es por fila
imputer = imputer.fit(X[:,1:3])
X[:,1:3] = imputer.transform(X[:,1:3])

File: DataScience/R/missing_data.r
#Plantilla para preprocesado de datos
#Importar dataset
dataset = read.csv('Data.csv')

#Tratamiento de datos faltantes NaN por medio de sustituirlos por la media
dataset$Age = ifelse(is.na(dataset$Age), 
                     ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)),
                     dataset$Age)

dataset$Salary = ifelse(is.na(dataset$Salary), 
                        ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
                        dataset$Salary)